### PR TITLE
Added Visibilty toggle for Rooms to home screen.

### PR DESCRIPTION
**Pull request by ShivamMenda**
Attempts to solve #88, Used the Visibility Widget to check whether there are rooms in the list or not and showed the appropriate widget to do so.

### CODE REVIEW

**Comment by chandansgowda** [comment]
@ShivamMenda Can you please a add a small screen recording of how it works?

**Comment by ShivamMenda** [comment]
@chandansgowda actually there's a sample room present in the room list already. And I'm not able to delete that. I just have a screenshot on how it looks with and without rooms, if that's okay? If you could remove that sample room and its empty initially then I could show you the flow where I add my own room and then it gets refreshed.

**Comment by ShivamMenda** [comment]
https://github.com/AOSSIE-Org/Resonate/assets/74780977/ac959e7f-651a-458a-9874-575684039cdc

**Comment by ShivamMenda** [comment]
@chandansgowda This is the flow, If the list is Empty, it'll return true and the NoRoom Widget will get triggered, and If there's a room it'll return false and the other widget will get triggered, As shown in the video it handles both cases.

**Review by chandansgowda — CHANGES_REQUESTED** [review]
Great job till here @ShivamMenda 
Need few changes-
1. Add any related illustration above No rooms (same colour scheme with less opacity)
2. Lets just try changing the Refresh button theme.. like make the border and text yellow and the bg transparent ?
3. Have you checked for responsiveness of the design ?

**Comment by ShivamMenda** [comment]
@chandansgowda Will make these changes and get back to you.

**Comment by ShivamMenda** [comment]
@chandansgowda I've made the changes requested and checked for responsiveness also, Use Get.height so everything is responsive.

**Comment by ShivamMenda** [comment]
Simulator Screenshot - iPhone 12 - 2023-08-01 at 08 06 12

**Comment by Aarush-Acharya** [comment]
This looks good

**Comment by ShivamMenda** [comment]
> This looks good 

@Aarush-Acharya Thank you! Glad you liked it!

**Comment by chandansgowda** [comment]
Great Job @ShivamMenda 
1. Can we have a refresh icon also inside the button ?  (make sure to keep it responsive)
2. Where did you get the illustration from? If the author has to be cited, please do so in our readme file.

**Comment by ShivamMenda** [comment]
@chandansgowda Made the requested changes. The illustration is from the Website https://undraw.co/. It is completely open source and does not require any attribution. So We are good to go!

**Comment by ShivamMenda** [comment]
Simulator Screenshot - iPhone 12 - 2023-08-01 at 14 48 23
Here's the final result

File: assets/images/NoRoom.png
Comment (chandansgowda) [inline assets/images/NoRoom.png:1]: @ShivamMenda Please use snake case for naming files

Comment (ShivamMenda) [reply assets/images/NoRoom.png:1]: Sure

**Comment by chandansgowda** [comment]
Also @ShivamMenda 
From next time, make sure to use our project's commit guidelines as mentioned in the Readme file.

**Comment by ShivamMenda** [comment]
@chandansgowda will be careful about it next time. Sorry.

**Review by chandansgowda — APPROVED** [review]
@jddeep This PR can be merged now. Please review from your end. Also please merge it as one commit as per our guidelines if possible.

**Pull request merged**